### PR TITLE
feat(GitOpsConfig): add .gitops.config.yaml apiVersion v1

### LINKED ISSUE #105: Refactor .gitops.config.yaml
Currently the `.gitops.config.yaml` file looks like this:
```yaml
deploymentConfig:
  # The organisation name of your deployment repo
  org: deployments
  # The repostiory name of your deployment repo
  repository: deployment-config-repo
  # The name of the application (name of the folder in `.preview-templates`)
  applicationName: app-xy

previewConfig:
  route:
    host:
      # Your router host
      # {SHA256_8CHAR_BRANCH_HASH} gets replaced by a shortened hash of your preview_id
      template: app.xy-{SHA256_8CHAR_BRANCH_HASH}.example.tld
  replace:
    # Paths that should be replaced in the `values.yaml`
    - path: image.tag
      variable: GIT_COMMIT # this is the git hash of your app repo
    - path: route.host
      variable: ROUTE_HOST # this is the resolved host.template from above
```
Proposed changes:
* Add an `apiVersion` field so we can introduce breaking changes (at least required for the proposed changes)
* Rename `deploymentConfig` to `configRepository` or `configRepo`
* Drop `previewConfig.route` as it is OpenShift specific and not flexible
* Add templating support to `previewConfig.replace[].variable` to make the template feature which currently lives under `previewConfig.route.host.template` available in a more flexible manner. Maybe we should drop `previewConfig.replace[].variable` as well and name it `previewConfig.replace[].value` with "${VAR}" support .
* Rename the available `SHA256_8CHAR_BRANCH_HASH` variable to `BRANCH_NAME_SHA_SHORT` or similar - the current name is not rememberable.

### PR DESCRIPTION
**Pull request by christiansiegel**
This resolves #105 

Introducing a new `.gitops.config.yaml` apiVersion with more flexibility. It is now possible to e.g.:
- Load preview templates from a repository other than the deployment repo (or just a different branch or folder)
- Template the namespace name (you can also include the branch name via `PREVIEW_ID` now)
- Template any yaml files (not just Chart.yaml and values.yaml) = non-Helm support

Simple config converter: https://christiansiegel.github.io/gitopscli-config-converter/

### CODE REVIEW

File: docs/commands/create-preview.md [hunk 1]
```diff
@@ -7,57 +7,67 @@ You need to provide some additional configuration files in your repositories for
...
 ### .gitops.config.yaml
 
 Make sure that your *app repository* contains a `.gitops.config.yaml` file. This file provides all information to 
 
-1. find the *deployment config repository*
-2. locate the preview template for your app
-3. replace image tag and route host in the template
+1. find repository, branch, and folder containing the template
+2. templates for host and namespace name
+3. replacements in template files
```
Comment (joachimprinzbach) [inline docs/commands/create-preview.md:32]: see top. not sure how to tackle the duplication in readme's

**Review by joachimprinzbach — APPROVED** [review]

File: gitopscli/gitops_config.py [hunk 2]
```diff
@@ -1,109 +1,322 @@
...
 
-        path: str
-        variable: Variable
+            self.path = path
+            self.value_template = value_template
 
-        def __post_init__(self) -> None:
-            assert isinstance(self.path, str), "path of wrong type!"
-            assert isinstance(self.variable, self.Variable), "variable of wrong type!"
+            for var in _VARIABLE_REGEX.findall(self.value_template):
```
Comment (niiku) [inline gitopscli/gitops_config.py:37]: Same code as in `__assert_variables`. I'm also not sure if an exception should be thrown, or the `{var}` should be simply left as it is.

**Review by niiku — APPROVED** [review]
Awesome stuff here! lgtm

File: docs/commands/create-preview.md (hunk 1 above)
Comment (christiansiegel) [reply docs/commands/create-preview.md]: markdown can be included with https://pypi.org/project/markdown-include/. I'll create another PR for that..

File: gitopscli/gitops_config.py [hunk 3]
```diff
@@ -1,109 +1,322 @@
...
 @dataclass(frozen=True)
 class GitOpsConfig:
-    @dataclass(frozen=True)
     class Replacement:
-        class Variable(Enum):
-            GIT_COMMIT = auto()
-            ROUTE_HOST = auto()
+        @dataclass(frozen=True)
+        class Context:
+            gitops_config: "GitOpsConfig"
```
[the remark this answers is not in the document]
Comment (christiansiegel) [reply gitopscli/gitops_config.py:17]: Forward declaration workaround. Details see https://www.python.org/dev/peps/pep-0484/#the-problem-of-forward-declarations

File: gitopscli/gitops_config.py (hunk 2 above)
Comment (christiansiegel) [reply gitopscli/gitops_config.py:37]: IMO the exception makes sense. If someone wants to have `{foo}` in the output as well, the brackets should be escaped in some way...e.g. `{{foo}}` (not implemented yet)..

**Pull request merged**